### PR TITLE
MGMT-7287: Monitor user selection of networkType by setting usage whe…

### PR DESCRIPTION
**Pull request by oshercc**
…n user select

# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @slaviered 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/usage/consts.go [hunk 1]
```diff
@@ -15,4 +15,6 @@ const (
 	VipDhcpAllocationUsage string = "VIP auto alloc."
 	//usage of disk selection
 	DiskSelectionUsage string = "Disk Selection"
+	//user networkType selection
+	NetworkTypeSelectionUsage = "NetworkType Selection"
```
Comment (gamli75) [inline internal/usage/consts.go:19]: you missed the string type and please change it to: Network Type Selection - this is the string that will appear in the dashboard

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -2037,6 +2037,10 @@ func (b *bareMetalInventory) updateNetworkParams(params installer.UpdateClusterP
 		setMachineNetworkCIDRForUpdate(updates, machineCidr)
 	}
 	if params.ClusterUpdateParams.NetworkType != nil && params.ClusterUpdateParams.NetworkType != cluster.NetworkType {
+		// If user
```
Comment (gamli75) [inline internal/bminventory/inventory.go:2040]: please remove this remark or complete the sentence

File: internal/usage/consts.go (hunk 1 above)
Comment (slaviered) [reply internal/usage/consts.go:19]: Should be NetworkType only. These are labels so the shortest description is preferable

**Comment by gamli75** [comment]
Please add test to https://github.com/openshift/assisted-service/blob/8e281ff3251ffac35ac7f58ee280678ae6e6a9ec/subsystem/cluster_test.go#L1095

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (slaviered) [reply internal/bminventory/inventory.go:2040]: No need for comment it is self explanatory

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -2037,6 +2037,10 @@ func (b *bareMetalInventory) updateNetworkParams(params installer.UpdateClusterP
...
+		b.setUsage(true, usage.NetworkTypeSelectionUsage, &map[string]interface{}{
+			"network_type": params.ClusterUpdateParams.NetworkType}, usages)
+
 		updates["network_type"] = swag.StringValue(params.ClusterUpdateParams.NetworkType)
```
Comment (slaviered) [inline internal/bminventory/inventory.go:2044]: is there a default network_type? If yes it should be added to setDefaultUsage() as well

**Review by slaviered — CHANGES_REQUESTED** [review]

File: internal/usage/consts.go [hunk 4]
```diff
@@ -15,4 +15,6 @@ const (
 	VipDhcpAllocationUsage string = "VIP auto alloc."
 	//usage of disk selection
 	DiskSelectionUsage string = "Disk Selection"
+	//user networkType selection
+	NetworkTypeSelectionUsage string = "NetworkType Selection"
```
Comment (gamli75) [inline internal/usage/consts.go:19]:
```suggestion
	NetworkTypeUsage string = "Network Type"
```

**Review by gamli75 — APPROVED** [review]

**Pull request merged**